sips/dynamic_qbft_timeout.md

```diff
@@ -0,0 +1,85 @@
+| Author         | Title                 | Category | Status   |
+| -------------- | --------------------- | -------- | -------- |
+| Gal Rogozinski |  QBFT timeout | Core     | approved |
+
+
+**Summary**  
+Changes to the constant timeouts we defined in [SIP#6](https://github.com/bloxapp/SIPs/blob/main/sips/constant_qbft_timeout.md) that should improve performance. 
+
+**Rational**  
+Currently for all duties, we have a `quickTimeout` of 2 seconds before sending a `RoundChange` message and a `slowTimeout` of 2 minutes starting from the 8th round.The timer starts when the QBFT instance starts. This approach has 2 issues:
+
+1. Duties that require a pre-consensus stage may start their instances at different times, causing some nodes to timeout sooner than others. This will result in a greater probability of a failed first round.
+2. Different duties have different timing assumptions, which may cause some duties to execute earlier than others.
+
+
+
+**Beacon Duty Timing Assumptions**  
+The Beacon chain has some timing assumptions regarding duty execution for maximizing rewards.
+
+| Duty           | Timing Assumption | Dependency     | Description                                                                                                         |
+| -------------- | ----------------- | -------------- | ------------------------------------------------------------------------------------------------------------------- |
+| Proposal       | < 4 Sec           | ---            | Attestations wait 4 seconds for a block                                                                             |
+| Attester       | < 4 Sec           | Proposal       | Attestations wait 4 seconds for a block then need to be propogated within 4 seconds for aggregators to pick them up |
+| Aggregator     | < 4 Sec           | Attester       | Aggregators start aggregating within 8 seconds from slot start and need to propogate within 4 seconds               |
+| Sync Committee | < 4 sec           | Proposal       | Sync Committee wait 4 seconds for a block                                                                           |
+| Contribution   | < 4 sec           | Sync Committee | Contributions wait 8 seconds from the slot start                                                                    |
+
+        
+
+**Specification**
+
+From the slot start time after `baseDuration` delay wait 2 seconds before emitting a `RoundChange` message until the 8th round (inclusive). Afterwards, wait 2 minutes.
+
+```go
+var (
+    quickTimeoutThreshold = Round(8)
```

Comment:
I am actually not sure why this is so high... 
On Round 6 the slot already ended

Reply:
6 is a more logical choice for more reasons even:
1. For 4 nodes committee we would have 2 nodes doing quick-timeouts twice
2. For 7 nodes committee, 6 nodes would do quick timeouts and the last node a slow one. But we managed to circle all the nodes.
3. For 13 nodes committee, 7 nodes would be more than half the committee...

Also later, not in this SIP, we may decide that there is no point to keep on doing consensus after the end of the slot, so this would be a preparation...

Reply:
@GalRogozinski
agreed, if this change would be a fork, then i'd suggest to also reduce `CutoffRound` to 1 slot for proposal and sync committee duties to reduce RC spam (afaik, they can't be retried after the slot has ended)